app/authentication/token.py

```diff
@@ -93,19 +88,34 @@ async def is_app_admin(token: str, app: str, error_str: str) -> bool:
         return True
 
 
-async def get_user(token: str = Depends(oauth2_scheme)) -> Tuple[str, str]:
-    """Calls GFW API to authorize user.
-
-    This functions check is user of any level is associated with the GFW
-    app and returns the user ID
-    """
+async def get_user(token: str = Depends(oauth2_scheme)) -> User:
+    """Get the details for authenticated user."""
 
     response: Response = await who_am_i(token)
 
-    if response.status_code == 401 or not (
-        "gfw" in response.json()["extraUserData"]["apps"]
-    ):
+    if response.status_code == 401:
         logger.info("Unauthorized user")
         raise HTTPException(status_code=401, detail="Unauthorized")
     else:
-        return response.json()["id"], response.json()["role"]
+        return User(**response.json()["data"])
```

Comment:
I may be missing something, but I'm not following this.  I don't see that you changed who_am_i(), so it is just returning the base level response from "/auth/check-logged".  But that doesn't have a "data" value.  It does have "email" and "extraUserData" and "role"  and "id", as shown by previous code in the file.  Won't the access to "data" just fail?

https://resource-watch.github.io/doc-api/reference.html#get-lt-base-api-url-gt-auth-check-logged

Reply:
@danscales  oh good catch, I copied it from the other line doing this, which calls `/auth/sign-up`. That returns it in a `data` object, but for some reason this endpoint doesn't. Which is weird, I thought that was just a standard (from `jsend`) across the entire RW API.  Fixed bow.